File: summer/WebContent/js/system/componentmodel/ITypeDescriptorContext.js
/**
 * ITypeDescriptorContext
 */

define(["dojo/_base/declare", "system/Type", "system/IServiceProvider"], 
		function(declare, Type, IServiceProvider){
	var ITypeDescriptorContext = declare("ITypeDescriptorContext", IServiceProvider,{
	});
	
	ITypeDescriptorContext.Type = new Type("ITypeDescriptorContext", ITypeDescriptorContext, [IServiceProvider.Type]);
	return ITypeDescriptorContext;
});



//	public interface ITypeDescriptorContext : IServiceProvider
//	{
//		IContainer Container
//		{
//			get;
//		}
//		object Instance
//		{
//			get;
//		}
//		PropertyDescriptor PropertyDescriptor
//		{
//			get;
//		}
//		bool OnComponentChanging();
//		void OnComponentChanged();
//	}

File: org.summer.view.widget/src/org/summer/view/widget/model/ITypeDescriptorContext.java
package org.summer.view.widget.model;

import org.summer.view.widget.IServiceProvider;

//[ComVisible (true)]
	public interface ITypeDescriptorContext extends IServiceProvider
	{
		IContainer Container { get; }

		Object Instance { get; }

		PropertyDescriptor PropertyDescriptor { get; }

		void OnComponentChanged ();

		boolean OnComponentChanging ();
	}